### PR TITLE
Add proxy flag and category to interfaces

### PR DESCRIPTION
**Pull request by schneidermic0**
Relates to #86.

### CODE REVIEW

**Comment by larshp** [comment]
for INTF I see `file-formats/intf/zif_aff_intf_v1.intf.abap` as being the source, and `file-formats/intf/intf.json` will sometime be auto-generated and overwritten?

ie. its most important to do the changes in the `.abap` file

**Comment by schneidermic0** [comment]
> for INTF I see file-formats/intf/zif_aff_intf_v1.intf.abap as being the source, and file-formats/intf/intf.json will sometime be auto-generated and overwritten?

Actually, I did it this way, but I mixed up files and did not recognise that I didn't add the changes to file-`formats/intf/zif_aff_intf_v1.intf.abap`

File: file-formats/intf/zif_aff_intf_v1.intf.abap [hunk 1]
```diff
@@ -1,5 +1,19 @@
 INTERFACE zif_aff_intf_v1 PUBLIC.
 
+  TYPES:
+    category_base TYPE n LENGTH 2,
```
[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/intf/zif_aff_intf_v1.intf.abap:4]: This is a false-positive, isn't it? The type is used in the ENUM below as base type

[the remark this answers is not in the document]
Comment (larshp) [reply file-formats/intf/zif_aff_intf_v1.intf.abap:4]: that looks like a false positive, I'll fix it

[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/intf/zif_aff_intf_v1.intf.abap:4]: Let me know, if I shall add an issue to abaplint

[the remark this answers is not in the document]
Comment (larshp) [reply file-formats/intf/zif_aff_intf_v1.intf.abap:4]: https://github.com/abaplint/abaplint/pull/1953

**Comment by larshp** [comment]
the editor config requires no newline at end of file, editing the file via github web always adds newline at end of file

File: file-formats/intf/zif_aff_intf_v1.intf.abap (hunk 1 above)
Comment (schneidermic0) [reply file-formats/intf/zif_aff_intf_v1.intf.abap:4]: @larshp Thanks

**Review by crasovan — APPROVED** [review]
ok

File: file-formats/intf/format.md [hunk 2]
```diff
@@ -8,4 +8,6 @@ Example files for an interface `zif_aff_example` are provided in the following t
 File | Content
  --- | ---
 [`zif_aff_example.intf.json`](./examples/zif_aff_example.intf.json)         | Properties and descriptions of the ABAP interface
-[`zif_aff_example.intf.abap`](./examples/zif_aff_example.intf.abap)         | Interface source code
\ No newline at end of file
+[`zif_aff_example.intf.abap`](./examples/zif_aff_example.intf.abap)         | Interface source code
+
+Documentation of interaces is not covered, yet.
```
Comment (larshp) [inline file-formats/intf/format.md:13]:
```suggestion
Documentation of interfaces is not covered, yet.
```

File: file-formats/intf/zif_aff_types_v1.intf.abap [hunk 3]
```diff
@@ -13,20 +13,15 @@ INTERFACE zif_aff_types_v1 PUBLIC.
     END OF ENUM abap_language_version.
 
   TYPES:
-    "! Head to be included in each object type
-    BEGIN OF head,
-      "! Location of the corresponding JSON schema
-      schema TYPE string,
```
Comment (Christopher-Hermann) [inline file-formats/intf/zif_aff_types_v1.intf.abap:19]: Why do you delete the schema in the abap type? It's still part of the json schema.

Comment (larshp) [reply file-formats/intf/zif_aff_types_v1.intf.abap:19]: versioning deferred to https://github.com/SAP/abap-file-formats/issues/53

the ABAP type is the source of truth going forward

**Comment by larshp** [comment]
conflicts resolved, lets merge this? and keep #86 open

File: file-formats/intf/zif_aff_types_v1.intf.abap (hunk 3 above)
Comment (schneidermic0) [reply file-formats/intf/zif_aff_types_v1.intf.abap:19]: I have just talked to @Christopher-Hermann. 
He is also convinced, now. I.e., we check whether we can get rid of the schema from the ABAP type. But as of now, we would keep it in the json-schema and the json.

**Comment by larshp** [comment]
github actions is kind of down, this is the 5th time over the last 7 days...

**Pull request merged**